src/genie/libs/parser/iosxr/show_rib.py

```diff
@@ -0,0 +1,200 @@
+                    
+                if group['forw_refe']=='N':
+                    result_dict[table_id].update({'forward_referenced': 'No'})
+                elif group['forw_refe']=='Y':
+                    result_dict[table_id].update({'forward_referenced': 'Yes'})
+                    
+                if group['tbl_del']=='N':
+                    result_dict[table_id].update({'table_deleted': 'No'})
+                elif group['tbl_del']=='Y':
+                    result_dict[table_id].update({'table_deleted': 'Yes'})
+
+                if group['tbl_conv']=='N':
+                    result_dict[table_id].update({'table_reached_convergence': 'No'})
+                elif group['tbl_conv']=='Y':
+                    result_dict[table_id].update({'table_reached_convergence': 'Yes'})                                
+        
+        result_dict = {'table_id': result_dict}
+        result_dict = {'rib_tables': result_dict}
+        return result_dict
+
+
+# =============================================
+# Parser for 'show rib tables summary'
+# =============================================
+
+class ShowRibTablesSummarySchema(MetaParser):
+    
+    schema = {'rib_summary': {
+                        'table_id':{
+                            
+                         Any(): {
+                            'num_unicast_tables': int,
+                            'total_unicast_prefixes': int,
+                            'num_multicast_tables': int,
+                            'total_multicast_prefixes': int,
+                        }
+                    }
+                }
+            }        
+
+
+
+class ShowRibTablesSummary(ShowRibTablesSummarySchema):
+    """ Parser for show rib tables summary"""
+
+    cli_command = ['show rib tables summary']
+
+    def cli(self,port="",output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+            
+        result_dict = {}
+        
+        p1 = re.compile(r'^Summary\sof.+in +(?P<rib>\S+)')
+        p2 = re.compile(r'^Number of unicast tables:\s+(?P<uni_tbl>\d+)')
+        p3 = re.compile(r'^Total number of unicast prefixes:\s+(?P<uni_pfx>\d+)')
+        p4 = re.compile(r'^Number of multicast tables:\s+(?P<multi_tbl>\d+)')
+        p5 = re.compile(r'^Total number of multicast prefixes:\s+(?P<multi_pfx>\d+)')
+
+        for line in out.splitlines():
+    
+            if line:
+                line = line.strip()
+
+            else:
+                continue
+                  
+            m = p1.match(line)
+            
+            if m:
+                group = m.groupdict()
+                table_id=group['rib'].lower()
+                result_dict[table_id]={}
```

Comment:
`result_dict.setdefault('table_id', {})`

Reply:
Corrected, as per your steer. result_dict.setdefault('rib_summary', {}).setdefault('table_id', {}) is created at the start.